=== modified file 'loggerhead/highlight.py'
--- old/loggerhead/highlight.py	2019-09-29 18:25:21 +0000
+++ new/loggerhead/highlight.py	2020-05-06 18:14:26 +0000
@@ -17,7 +17,10 @@
 #
 
 import breezy.osutils
-import cgi
+try:
+    from html import escape
+except ImportError:
+    from cgi import escape
 
 from pygments import highlight as _highlight_func
 from pygments.lexers import guess_lexer, guess_lexer_for_filename, TextLexer
@@ -37,7 +40,7 @@
     """
 
     if len(text) > MAX_HIGHLIGHT_SIZE:
-        return map(cgi.escape,  breezy.osutils.split_lines(text))
+        return map(escape,  breezy.osutils.split_lines(text))
 
     formatter = HtmlFormatter(style=style, nowrap=True, classprefix='pyg-')
 

=== modified file 'loggerhead/tests/test_simple.py'
--- old/loggerhead/tests/test_simple.py	2019-09-29 18:25:21 +0000
+++ new/loggerhead/tests/test_simple.py	2020-05-06 18:14:53 +0000
@@ -17,7 +17,10 @@
 
 from __future__ import absolute_import
 
-import cgi
+try:
+    from html import escape
+except ImportError:
+    from cgi import escape
 import logging
 import re
 import simplejson
@@ -94,12 +97,12 @@
     def test_changes(self):
         app = self.setUpLoggerhead()
         res = app.get('/changes')
-        res.mustcontain(cgi.escape(self.msg))
+        res.mustcontain(escape(self.msg))
 
     def test_changes_for_file(self):
         app = self.setUpLoggerhead()
         res = app.get('/changes?filter_file_id=myfilename-id')
-        res.mustcontain(cgi.escape(self.msg))
+        res.mustcontain(escape(self.msg))
 
     def test_changes_branch_from(self):
         app = self.setUpLoggerhead(served_url="lp:loggerhead")
@@ -126,7 +129,7 @@
         body_no_span = re.sub(b'<span class="pyg-.">', b'', res.body)
         body_no_span = body_no_span.replace(b'</span>', b'')
         for line in self.filecontents.splitlines():
-            escaped = cgi.escape(line).encode('utf-8')
+            escaped = escape(line).encode('utf-8')
             self.assertTrue(escaped in body_no_span,
                             "did not find %r in %r" % (escaped, body_no_span))
 

