Origin: commit, revision id: jelmer@jelmer.uk-20200604194550-uodf818ule2x2kxo
Author: Jelmer Vernooĳ <jelmer@jelmer.uk>
Last-Update: 2020-06-04
X-Bzr-Revision-Id: jelmer@jelmer.uk-20200604194550-uodf818ule2x2kxo

=== modified file '.travis.yml'
--- old/.travis.yml	2018-11-01 22:25:03 +0000
+++ new/.travis.yml	2020-06-04 19:43:36 +0000
@@ -20,7 +20,7 @@
   - python -m coverage run -p -m unittest loggerhead.tests.test_suite
 
 install:
-  - sudo apt install python-all subunit python-testtools adduser libjs-yui3-min
+  - sudo apt install python-all subunit python-testtools adduser libjs-jquery
   - travis_retry pip install -U setuptools
   - travis_retry pip install -U pip coverage codecov flake8 testtools configobj cython testscenarios six docutils python-subunit dulwich bzr+lp:brz pygments simplejson paste http://www.owlfish.com/software/simpleTAL/downloads/SimpleTAL-$TAL_VERSION.tar.gz bleach
 

=== modified file 'byov.conf'
--- old/byov.conf	2018-11-04 18:26:11 +0000
+++ new/byov.conf	2020-06-04 19:43:36 +0000
@@ -11,7 +11,7 @@
 [loggerhead]
 vm.release = xenial
 brz.build_deps = gcc, debhelper, python, python-all-dev, python3-all-dev, python-configobj, python3-configobj, python-docutils, python3-docutils, python-paramiko, python3-paramiko, python-subunit, python3-subunit, python-testtools, python3-testtools, subunit, cython, cython3, python-fastimport, python-dulwich, python-six, python3-six
-loggerhead.build_deps = python-setuptools, python3-setuptools, libjs-yui3-min, python-docutils, python3-docutils, python-pygments, python3-pygments, python-simplejson, python3-simplejson, python-paste, python3-paste, python-pastedeploy, python3-pastedeploy, python-simpletal, python-bleach, python3-bleach
+loggerhead.build_deps = python-setuptools, python3-setuptools, libjs-jquery, python-docutils, python3-docutils, python-pygments, python3-pygments, python-simplejson, python3-simplejson, python-paste, python3-paste, python-pastedeploy, python3-pastedeploy, python-simpletal, python-bleach, python3-bleach
 loggerhead.test_deps = python3-fixtures, python-fixtures
 vm.packages = {brz.build_deps}, {loggerhead.build_deps}, {loggerhead.test_deps}, bzr, python-junitxml, python3-pip
 brz.branch = (bzr branch lp:brz ../brz-trunk)

=== modified file 'docs/loggerhead-serve.rst'
--- old/docs/loggerhead-serve.rst	2018-10-20 17:34:46 +0000
+++ new/docs/loggerhead-serve.rst	2020-06-04 19:43:36 +0000
@@ -72,7 +72,7 @@
 
 .. cmdoption:: --use-cdn
    
-    Serve YUI javascript libraries from Yahoo!'s CDN.
+    Serve jQuery javascript libraries from Googles CDN.
 
 .. cmdoption:: --allow-writes
    

=== modified file 'loggerhead/apps/branch.py'
--- old/loggerhead/apps/branch.py	2020-05-06 19:18:04 +0000
+++ new/loggerhead/apps/branch.py	2020-06-04 19:43:36 +0000
@@ -134,12 +134,13 @@
     def static_url(self, path):
         return self._static_url_base + path
 
-    def yui_url(self, path):
+    def js_library_url(self, path):
         if self.use_cdn:
-            base = 'http://yui.yahooapis.com/3.0.0pr2/build/'
+            if path == 'jquery.min.js':
+                return 'https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js'
+            raise KeyError('unknown js library %s' % path)
         else:
-            base = self.static_url('/static/javascript/yui/build/')
-        return base + path
+            return self.static_url('/static/javascript/' + path)
 
     controllers_dict = {
         '+filediff': FileDiffUI,

=== modified file 'loggerhead/config.py'
--- old/loggerhead/config.py	2018-09-08 15:16:32 +0000
+++ new/loggerhead/config.py	2020-06-04 19:43:36 +0000
@@ -71,7 +71,7 @@
     parser.add_option("--version", action="store_true", dest="show_version",
                       help="Print the software version and exit")
     parser.add_option("--use-cdn", action="store_true", dest="use_cdn",
-                      help="Serve YUI from Yahoo!'s CDN")
+                      help="Serve jQuery from Google's CDN")
     parser.add_option("--cache-dir", dest="sql_dir",
                       help="The directory to place the SQL cache in")
     parser.add_option("--allow-writes", action="store_true",

=== modified file 'loggerhead/static/javascript/changelog.js'
--- old/loggerhead/static/javascript/changelog.js	2009-03-31 21:11:38 +0000
+++ new/loggerhead/static/javascript/changelog.js	2020-06-04 19:43:36 +0000
@@ -1,70 +1,63 @@
 function toggle_expand_all(action)
 {
-  var revlogs = Y.all('.revision_log');
-  if (revlogs == null) return;
-  revlogs.each(
-    function(item, i)
+  $('.revision_log').each(
+    function(i, item)
     {
       var collapsable = item.collapsable;
       if(action == 'close')
       {
-        Y.get('#expand_all').setStyle('display','block');
-        Y.get('#collapse_all').setStyle('display','none');
+        $('#expand_all').css({'display': 'block'});
+        $('#collapse_all').css({'display': 'none'});
         collapsable.close();
       }
       else if(action == 'open')
       {
-        Y.get('#expand_all').setStyle('display','none');
-        Y.get('#collapse_all').setStyle('display','block');
+        $('#expand_all').css({'display': 'none'});
+        $('#collapse_all').css({'display': 'block'});
         collapsable.open();
       }
     });
 }
 
-Y.on(
-  'click',
+$(function() {
+$('#expand_all a').on('click',
   function (event) {
     event.preventDefault();
     toggle_expand_all('open');
   },
-  '#expand_all a'
 );
+});
 
-Y.on(
-  'click',
+$(function() {
+$('#collapse_all a').on('click',
   function (event) {
     event.preventDefault();
     toggle_expand_all('close');
   },
-  '#collapse_all a'
 );
-
-Y.on(
-  "domready", function () {
-    Y.all(".show_if_js").removeClass("show_if_js");
-  });
-
-Y.on(
-  'domready',
-  function()
+});
+
+$(function () {
+    $(".show_if_js").removeClass("show_if_js");
+});
+
+$(function()
   {
-    var revlogs = Y.all('.revision_log');
-    if (revlogs == null) return;
-    revlogs.each(
-      function(item, i)
+    $('.revision_log').each(
+      function(i, item)
       {
-        var revid = revids[item.get('id').replace('log-', '')];
+        var revid = revids[item.id.replace('log-', '')];
         var collapsable = new Collapsable(
           {
-            expand_icon: item.query('.expand_icon'),
-            open_node: item.query('.long_description'),
-            close_node: item.query('.short_description'),
+            expand_icon: $(item).find('.expand_icon'),
+            open_node: $(item).find('.long_description'),
+            close_node: $(item).find('.short_description'),
             source: global_path + '+revlog/' + revid,
-            loading: item.query('.loading'),
+            loading: $(item).find('.loading'),
             is_open: false
           });
 
-        item.query('.expand_revisioninfo a').on(
+        $(item).find('.expand_revisioninfo a').on(
           'click',
           function(e) {
             e.preventDefault();

=== modified file 'loggerhead/static/javascript/custom.js'
--- old/loggerhead/static/javascript/custom.js	2011-10-26 16:06:54 +0000
+++ new/loggerhead/static/javascript/custom.js	2020-06-04 19:43:36 +0000
@@ -1,47 +1,37 @@
-Y = YUI().use("base", "node", "io-base", "anim");
-
 var global_timeout_id = null;
 var global_search_request = null;
 
-Y.on(
-  'domready',
-  function()
+$(function()
   {
-    var search_box = Y.get('#q');
-    if (!Y.Lang.isNull(search_box))
+    var search_box = $('#q');
+    if (search_box != undefined)
     {
       function get_suggestions() {
-        var query = search_box.get('value');
+        var query = search_box.value;
         var url = global_path + 'search?query=' + query;
 
-        if (!Y.Lang.isNull(global_search_request))
+        if (global_search_request != undefined)
         {
           global_search_request.abort();
         }
-        global_search_request = Y.io(
-          url,
-          {
-            on: {complete: cool_search},
-            arguments: [query]
-          }
-        );
-
-        var region = search_box.get('region');
-        var current_query = search_box.get('value');
-
-        Y.get('#search_terms').setStyle('display', 'block');
-        Y.get('#search_terms').setStyle('position', 'absolute');
-        Y.get('#search_terms').setStyle('left', region.left);
-        Y.get('#search_terms').setStyle('top', region.bottom);
-        Y.get('#search_terms').set('innerHTML','Loading...');
-      }
+        global_search_request = $.get(url, {'query': query}).done(cool_search);
+
+        var current_query = search_box.value;
+
+        $('#search_terms').css({
+            'display': 'block',
+            'position': 'absolute',
+            'left': search_box.left,
+            'top': search_box.bottom,
+            'innerHTML': 'Loading...',
+      });
       search_box.on(
         "keyup",
         function(event)
         {
-          if(search_box.get('value') == '')
+          if(search_box.value == '')
           {
-            Y.get('#search_terms').setStyle('display', 'none');
+            $('#search_terms').css({'display': 'none'});
           }
           else
           {
@@ -52,27 +42,29 @@
             global_timeout_id = setTimeout(get_suggestions, 200);
           }
         });
+      }
     }
-  });
+});
 
 function cool_search(tid, response, query)
 {
-  var q = Y.get('#q');
-  var region = q.get('region');
-  var current_query = q.get('value');
+  var q = $('#q');
+  var current_query = q.value;
   if (current_query == query)
   {
-    Y.get('#search_terms').set('innerHTML', response.responseText);
-    Y.get('#search_terms').setStyle('display', 'block');
-    Y.get('#search_terms').setStyle('position', 'absolute');
-    Y.get('#search_terms').setStyle('left', region.left);
-    Y.get('#search_terms').setStyle('top', region.bottom);
+    $('#search_terms').html(response.responseText);
+    $('#search_terms').css({
+       'display': 'block',
+       'position': 'absolute',
+       'left': q.left,
+       'top': q.bottom,
+    });
   }
 }
 
 function hide_search()
 {
-  setTimeout("Y.get('#search_terms').setStyle('display','none')", 300);
+  setTimeout("$('#search_terms').css({'display': 'none'})", 300);
 }
 
 function Collapsable(config)
@@ -90,13 +82,17 @@
 }
 
 function get_height(node) {
-  node.setStyle('position', 'absolute');
-  node.setStyle('top', -1000000000);
-  node.setStyle('display', 'block');
-  var height = node.get('region').height;
-  node.setStyle('display', 'none');
-  node.setStyle('position', 'static');
-  node.setStyle('top', 'auto');
+  $(node).css({
+     'position': 'absolute',
+     'top': -1000000000,
+     'display': 'block',
+  });
+  var height = $(node).height;
+  $(node).css({
+     'display': 'none',
+     'position': 'static',
+     'top': 'auto',
+  });
   return height;
 }
 
@@ -104,43 +100,36 @@
 {
   if (this.anim) this.anim.stop();
 
-  this.anim = new Y.Anim(
-    {
-      node: this.container,
-      from: {
-        marginBottom: this.container.getStyle('marginBottom')
-      },
-      to: {
-        marginBottom: 0
-      },
-      duration: 0.2
-    });
-
-  this.anim.run();
-  this.anim.on('end', this.animComplete, this, callback);
+  this.anim = $(this.container).animate(
+     {marginBottom: 0 }, 0.2, "swing" , function(event) {
+       this.anim = null;
+       if (this._loading) return;
+       if (callback) callback();
+       this.is_open = true;
+     });
 }
 
-Collapsable.prototype._load_finished = function(tid, res, args)
+Collapsable.prototype._load_finished = function(data, callback)
 {
-  var l = res.responseText.split('\n');
+  var l = data.split('\n');
   l.splice(0, 1);
-  var newNode = Y.Node.create(l.join(''));
+  var newNode = $(l.join(''));
   if (this.node_process)
     this.node_process(newNode);
   this.source = null;
-  newNode.setStyle('display', 'none');
-  this.loading.ancestor().insertBefore(newNode, this.loading);
-  var delta = this.loading.get('region').height - get_height(newNode);
-  newNode.setStyle('display', 'block');
-  this.container.setStyle('marginBottom', parseFloat(this.container.getStyle('marginBottom')) + delta);
-  this.loading.ancestor().removeChild(this.loading);
-  this._animate(args[0]);
+  newNode.css({'display': 'none'});
+  newNode.insertBefore(this.loading);
+  var delta = this.loading.height - get_height(newNode);
+  newNode.css({'display': 'block'});
+  this.container.css({'marginBottom': parseFloat(this.container.css('marginBottom')) + delta});
+  this.loading.remove();
+  this._animate(callback);
 };
 
 Collapsable.prototype._ensure_container = function(callback)
 {
   if (this.container == null) {
-    this.container = Y.Node.create('<div></div>');
+    this.container = $('<div></div>');
     if (this.closed_node) {
       this.closed_node.ancestor().replaceChild(
         this.container, this.closed_node);
@@ -150,13 +139,12 @@
       }
     }
     else {
-      this.open_node.ancestor().replaceChild(
-        this.container, this.open_node);
-      this.container.appendChild(this.open_node);
+      $(this.open_node).replaceWith(this.container);
+      $(this.container).append(this.open_node);
     }
-    var outer = Y.Node.create('<div style="overflow:hidden;"></div>');
-    this.container.ancestor().replaceChild(outer, this.container);
-    outer.appendChild(this.container);
+    var outer = $('<div style="overflow:hidden;"></div>');
+    this.container.replaceWith(outer);
+    $(outer).append(this.container);
   }
 }
 
@@ -186,7 +174,7 @@
 
 Collapsable.prototype.open = function(callback)
 {
-  this.expand_icon.set('src', expanded_icon_path);
+  this.expand_icon.src = expanded_icon_path;
 
   this._ensure_container();
 
@@ -194,46 +182,36 @@
 
   var close_height;
   if (this.close_node) {
-    close_height = this.close_node.get('region').height;
+    close_height = this.close_node.height;
   }
   else {
     close_height = 0;
   }
 
-  this.container.setStyle('marginBottom', close_height - open_height);
+  $(this.container).css({'marginBottom': close_height - open_height});
   if (this.close_node) {
-    this.close_node.setStyle('display', 'none');
+    $(this.close_node).css({'display': 'none'});
   }
-  this.open_node.setStyle('display', 'block');
+  $(this.open_node).css({'display': 'block'});
 
   this._animate(callback);
 
+  var collapsable = this;
+
   if (this.source) {
-    Y.io(
-      this.source,
-      {
-        on: {complete: this._load_finished},
-        arguments: [callback],
-        context: this
+      $.get(this.source, function(data) {
+            collapsable._load_finished(data, callback);
       });
     return;
   }
 
 };
 
-Collapsable.prototype.animComplete = function(evt, callback)
-{
-  this.anim = null;
-  if (this._loading) return;
-  if (callback) callback();
-  this.is_open = true;
-};
-
 Collapsable.prototype.close = function()
 {
   this._ensure_container();
 
-  var open_height = this.open_node.get('region').height;
+  var open_height = this.open_node.height;
 
   var close_height;
   if (this.close_node) {
@@ -243,27 +221,17 @@
     close_height = 0;
   }
 
-  var anim = new Y.Anim(
-    {
-      node: this.container,
-      from: {
-        marginBottom: 0
-      },
-      to: {
-        marginBottom: close_height - open_height
-      },
-      duration: 0.2
-    });
-  anim.on("end", this.closeComplete, this);
-  anim.run();
+  var anim = $(this.container).animate(
+      { marginBottom: [0, close_height - open_height]},
+      0.2, "swing", this.closeComplete);
 };
 
 Collapsable.prototype.closeComplete = function () {
-  this.open_node.setStyle('display', 'none');
+  $(this.open_node).css({'display': 'none'});
   if (this.close_node) {
-    this.close_node.setStyle('display', 'block');
+    $(this.close_node).css({'display': 'block'});
   }
-  this.container.setStyle('marginBottom', 0);
+  $(this.container).css({'marginBottom': 0});
   this.expand_icon.set('src', collapsed_icon_path);
   this.is_open = false;
 };
@@ -305,13 +273,13 @@
         }
     }
     var id_selector = "#" + target_id;
-    var main = Y.get(id_selector);
-    notification_node = Y.Node.create('<div></div>')
+    var main = $(id_selector);
+    notification_node = $('<div></div>')
         .addClass('global-notification');
     if (hidden) {
         notification_node.addClass('hidden');
     }
-    var notification_span = Y.Node.create('<span></span>')
+    var notification_span = $('<span></span>')
         .addClass('sprite')
         .addClass('notification-private');
     notification_node.set('innerHTML', notification_text);
@@ -324,43 +292,29 @@
      this is because we have no way to use feature flags in
      css directly. This should be removed if the feature
      is accepted. */
-    var body = Y.get('body');
+    var body = $(document.body);
     body.addClass('feature-flag-bugs-private-notification-enabled');
     // Set the visible flag so that the content moves down.
     body.addClass('global-notification-visible');
 
     setup_privacy_notification();
-    var global_notification = Y.get('.global-notification');
+    var global_notification = $('.global-notification');
     if (global_notification.hasClass('hidden')) {
         global_notification.addClass('transparent');
         global_notification.removeClass('hidden');
 
-        var fade_in = new Y.Anim({
-            node: global_notification,
-            to: {opacity: 1},
-            duration: 0.3
-        });
-        var body_space = new Y.Anim({
-            node: 'body',
-            to: {'paddingTop': '40px'},
-            duration: 0.2,
-            easing: Y.Easing.easeOut
-        });
-        var black_link_space = new Y.Anim({
-            node: '.black-link',
-            to: {'top': '45px'},
-            duration: 0.2,
-            easing: Y.Easing.easeOut
-        });
-
-        fade_in.run();
-        body_space.run();
-        black_link_space.run();
+        $(global_notification).animate({
+            opacity: 1,
+        }, 0.3);
+        $(document.body).animate(
+            {'paddingTop': '40px'}, 0.2, "easeOutBounce");
+        $('.black-link').animate(
+            {'top': '45px'}, 0.2, "easeOutBounce");
     }
 };
 
-Y.on('domready', function() {
-    var body = Y.get('body');
+$(function() {
+    var body = $(document.body);
     if (body.hasClass('private')) {
         setup_privacy_notification();
         display_privacy_notification();

=== modified file 'loggerhead/static/javascript/diff.js'
--- old/loggerhead/static/javascript/diff.js	2012-07-31 20:25:42 +0000
+++ new/loggerhead/static/javascript/diff.js	2020-06-04 19:43:36 +0000
@@ -4,32 +4,32 @@
   var pending_added = [];
   function flush_adds(before) {
     for (var i = 0; i < pending_added.length; i++) {
-      before.ancestor().insertBefore(pending_added[i], before);
+      $(pending_added[i]).insertBefore(before);
     }
     pending_added.length = 0;
   }
   chunk.get('children').filter(".pseudorow").each(
-    function (line) {
+    function (i, line) {
       if (line.hasClass("context-row")) {
         flush_adds(line);
-        line.removeChild(line.query('.code'));
+        line.removeChild($(line).find('.code'));
       }
       else if (line.hasClass("both-row")) {
         var added_line = line.create('<div class="pseudorow insert-row"><div class="lineNumber first">&nbsp;</div><div class="clear">&nbsp;</div></div>');
-        var clear = added_line.query('.clear');
-        added_line.insertBefore(line.query('.lineNumber.second'), clear);
-        added_line.insertBefore(line.query('.code.insert'), clear);
+        var clear = $(added_line).find('.clear');
+        $(line).find('.lineNumber.second').insertBefore(clear);
+        $(line).find('.code.insert').insertBefore(clear);
         pending_added[pending_added.length] = added_line;
-        line.insertBefore(line.create('<div class="lineNumber second">&nbsp;</div>'), line.query('.code.delete'));
+        $('<div class="lineNumber second">&nbsp;</div>').insertBefore($(line).find('.code.delete'));
         line.replaceClass("both-row", "delete-row");
       }
       else if (line.hasClass("insert-row")) {
         flush_adds(line);
-        line.removeChild(line.query('.blank'));
+        line.removeChild($(line).find('.blank'));
       }
       else if (line.hasClass("delete-row")) {
-        line.removeChild(line.query('.blank'));
-        line.insertBefore(line.query('.lineNumber.second'), line.query('.code.delete'));
+        $(line).find('.blank').remove();
+        $(line).find('.lineNumber.second').insertBefore($(line).find('.code.delete'));
       }
     });
   flush_adds(null);
@@ -46,32 +46,32 @@
       var a = added[i];
       var r = removed[i];
       a.ancestor().removeChild(a);
-      r.removeChild(r.query('.lineNumber.second'));
-      r.insertBefore(a.query('.lineNumber.second'), r.query('.clear'));
-      r.insertBefore(a.query('.code.insert'), r.query('.clear'));
+      r.removeChild($(r).find('.lineNumber.second'));
+      $(a).find('.lineNumber.second').insertBefore($(r).find('.clear'));
+      $(a).find('.code.insert').insertBefore($(r).find('.clear'));
       r.replaceClass('removed-row', 'both-row');
     }
     if (added.length > removed.length) {
       for (var j = common; j < added.length; j++) {
-        a = added[j];
-        a.insertBefore(a.create('<div class="code blank">&nbsp;</div>'), a.query('.lineNumber.second'));
+        a = $(added[j]);
+        $('<div class="code blank">&nbsp;</div>').insertBefore(a.find('.lineNumber.second'));
       }
     }
     else if (added.length < removed.length) {
       for (var j = common; j < removed.length; j++) {
-        r = removed[j];
-        r.insertBefore(r.query('.code.delete'), r.query('.lineNumber.second'));
-        r.insertBefore(r.create('<div class="code blank">&nbsp;</div>'), r.query('.clear'));
+        r = $(removed[j]);
+        r.find('.code.delete').insertBefore(r.find('.lineNumber.second'));
+        r.create('<div class="code blank">&nbsp;</div>').insertBefore(r.find('.clear'));
       }
     }
     added.length = 0;
     removed.length = 0;
   }
   chunk.get('children').filter(".pseudorow").each(
-    function (line) {
+    function (i, line) {
       if (line.hasClass("context-row")) {
         clear_bufs(line);
-        line.insertBefore(line.query('.code').cloneNode(true), line.query(".second"));
+        $(line).find('.code').cloneNode(true).insertBefore($(line).find(".second"));
       }
       else if (line.hasClass("insert-row")) {
         added[added.length] = line;
@@ -87,24 +87,26 @@
 
 function toggle_unified_sbs(event) {
   event.preventDefault();
-  var pts = Y.all(".pseudotable");
+  var pts = $(".pseudotable");
   if (unified) {
     pts && pts.each(make_sbs);
     unified = false;
-    Y.get("#toggle_unified_sbs").set('innerHTML', "Show unified diffs");
+    $("#toggle_unified_sbs").set('innerHTML', "Show unified diffs");
   }
   else {
     pts && pts.each(make_unified);
     unified = true;
-    Y.get("#toggle_unified_sbs").set('innerHTML', "Show diffs side-by-side");
+    $("#toggle_unified_sbs").set('innerHTML', "Show diffs side-by-side");
   }
 }
 
-Y.on("click", toggle_unified_sbs, '#toggle_unified_sbs');
+$(function() {
+$('#toggle_unified_sbs').on("click", toggle_unified_sbs);
+});
 
 function toggle_expand_all_revisionview(action)
 {
-  var diffs = Y.all('.diff');
+  var diffs = $('.diff');
   if (diffs == null) return;
   diffs.each(
     function(item, i)
@@ -112,36 +114,38 @@
       var collapsable = item.collapsable;
       if(action == 'close')
       {
-        Y.get('#expand_all').setStyle('display','block');
-        Y.get('#collapse_all').setStyle('display','none');
+        $('#expand_all').css({'display': 'block'});
+        $('#collapse_all').css({'display': 'none'});
         collapsable.close();
       }
       else if(action == 'open')
       {
-        Y.get('#expand_all').setStyle('display','none');
-        Y.get('#collapse_all').setStyle('display','block');
+        $('#expand_all').css({'display': 'none'});
+        $('#collapse_all').css({'display': 'block'});
         collapsable.open();
       }
     });
 }
 
-Y.on(
+$(function() {
+$('#expand_all a').on(
   'click',
   function (event) {
     event.preventDefault();
     toggle_expand_all_revisionview('open');
   },
-  '#expand_all a'
 );
+});
 
-Y.on(
+$(function() {
+$('#collapse_all a').on(
   'click',
   function (event) {
     event.preventDefault();
     toggle_expand_all_revisionview('close');
   },
-  '#collapse_all a'
 );
+});
 
 function node_process(node) {
   if (!unified) {
@@ -150,7 +154,7 @@
 }
 
 function zoom_to_diff (path) {
-  var collapsable = Y.get('#' + path_to_id[path]).collapsable;
+  var collapsable = $('#' + path_to_id[path]).collapsable;
   if (!collapsable.is_open) {
     collapsable.open(
       function () {
@@ -162,58 +166,55 @@
 var original_diff_download_link = null;
 
 function compute_diff_links() {
-  var diffs = Y.all('.diff');
-  if (diffs == null) return;
-  var numlines = document.getElementById('contextLines').value;
-  diffs.each(
-    function(item, i)
+  var numlines = $('#contextLines').value;
+  $('.diff').each(
+    function(i, item)
     {
-      item.collapsable.source = global_path + '+filediff/' + link_data[item.get('id')] + '?context=' + numlines; 
+      item.collapsable.source = global_path + '+filediff/' + link_data[item.id] + '?context=' + numlines;
     });
-  if(original_diff_download_link == null) original_diff_download_link = document.getElementById('download_link').href;
-  document.getElementById('download_link').href = original_diff_download_link + '?context=' + numlines;
+  if(original_diff_download_link == null) original_diff_download_link = $('#download_link').href;
+  $('#download_link').href = original_diff_download_link + '?context=' + numlines;
 }
 
 function get_num_lines() {
-  var numlines = document.getElementById('contextLines').value;
+  var numlines = $('#contextLines').value;
   return numlines;
 }
 
-Y.on(
-  "domready", function () {
-    Y.all(".show_if_js").removeClass("show_if_js");
+$(function () {
+    $(".show_if_js").removeClass("show_if_js");
     if (!specific_path) {
-      Y.all("#list-files a").on(
+      $("a#list-files").on(
         'click',
         function (e) {
           e.preventDefault();
-          var path = decodeURIComponent(e.target.get('href').split('#')[1]);
+          var path = decodeURIComponent(e.target.href.split('#')[1]);
           window.location.hash = '#' + path;
           zoom_to_diff(path);
         });
     }
-    var diffs = Y.all('.diff');
+    var diffs = $('.diff');
     if (diffs == null) return;
-    var numlines = document.getElementById('contextLines').value;
+    var numlines = $('#contextLines').value;
     diffs.each(
-      function(item, i)
+      function(i, item)
       {
-        var source_url = global_path + '+filediff/' + link_data[item.get('id')] + '?context=' + numlines;
-        item.query('.the-link').on(
+        var source_url = global_path + '+filediff/' + link_data[item.id] + '?context=' + numlines;
+        $(item).find('.the-link').on(
           'click',
           function(e) {
             e.preventDefault();
-            item.collapsable.source = global_path + '+filediff/' + link_data[item.get('id')] + '?context=' + document.getElementById('contextLines').value;
+            item.collapsable.source = global_path + '+filediff/' + link_data[item.id] + '?context=' + $('#contextLines').value;
             collapsable.toggle();
           });
         var collapsable = new Collapsable(
           {
-            expand_icon: item.query('.expand_diff'),
-            open_node: item.query('.diffinfo'),
+            expand_icon: $(item).find('.expand_diff'),
+            open_node: $(item).find('.diffinfo'),
             close_node: null,
             source: source_url,
             is_open: specific_path != null,
-            loading: item.query('.loading'),
+            loading: $(item).find('.loading'),
             node_process: node_process
           });
        item.collapsable=collapsable;

=== modified file 'loggerhead/templates/macros.pt'
--- old/loggerhead/templates/macros.pt	2018-09-08 14:54:01 +0000
+++ new/loggerhead/templates/macros.pt	2020-06-04 19:43:36 +0000
@@ -21,23 +21,7 @@
       var expanded_icon_path = <tal:block content="python:'\''+branch.static_url('/static/images/treeExpanded.png')+'\''" />;
     </script>
     <script type="text/javascript"
-            tal:attributes="src python:branch.yui_url('yui/yui-min.js')"></script>
-    <script type="text/javascript"
-            tal:attributes="src python:branch.yui_url('oop/oop-min.js')"></script>
-    <script type="text/javascript"
-            tal:attributes="src python:branch.yui_url('event/event-min.js')"></script>
-    <script type="text/javascript"
-            tal:attributes="src python:branch.yui_url('attribute/attribute-min.js')"></script>
-    <script type="text/javascript"
-            tal:attributes="src python:branch.yui_url('base/base-min.js')"></script>
-    <script type="text/javascript"
-            tal:attributes="src python:branch.yui_url('dom/dom-min.js')"></script>
-    <script type="text/javascript"
-            tal:attributes="src python:branch.yui_url('node/node-min.js')"></script>
-    <script type="text/javascript"
-            tal:attributes="src python:branch.yui_url('anim/anim-min.js')"></script>
-    <script type="text/javascript"
-            tal:attributes="src python:branch.yui_url('io/io-base-min.js')"></script>
+            tal:attributes="src python:branch.js_library_url('jquery.min.js')"></script>
     <script type="text/javascript"
             tal:attributes="src python:branch.static_url('/static/javascript/custom.js')"></script>
     <metal:block metal:define-slot="header_extras" />

=== modified file 'setup.py'
--- old/setup.py	2019-10-01 12:31:50 +0000
+++ new/setup.py	2020-06-04 19:43:36 +0000
@@ -49,28 +49,6 @@
     package_data = {"loggerhead": ["templates/*.pt",
                                    "static/css/*.css",
                                    "static/javascript/*.js",
-                                   "static/javascript/yui/build/anim/*",
-                                   "static/javascript/yui/build/base/*",
-                                   "static/javascript/yui/build/cssbase/*",
-                                   "static/javascript/yui/build/cssgrids/*",
-                                   "static/javascript/yui/build/dd/*",
-                                   "static/javascript/yui/build/dump/*",
-                                   "static/javascript/yui/build/get/*",
-                                   "static/javascript/yui/build/json/*",
-                                   "static/javascript/yui/build/node/*",
-                                   "static/javascript/yui/build/queue/*",
-                                   "static/javascript/yui/build/yui/*",
-                                   "static/javascript/yui/build/attribute/*",
-                                   "static/javascript/yui/build/cookie/*",
-                                   "static/javascript/yui/build/cssfonts/*",
-                                   "static/javascript/yui/build/cssreset/*",
-                                   "static/javascript/yui/build/dom/*",
-                                   "static/javascript/yui/build/event/*",
-                                   "static/javascript/yui/build/io/*",
-                                   "static/javascript/yui/build/loader/*",
-                                   "static/javascript/yui/build/oop/*",
-                                   "static/javascript/yui/build/substitute/*",
-                                   "static/javascript/yui/build/yui-base/*",
                                    "static/images/*"]},
     data_files = [
         ('share/man/man1', ['loggerhead-serve.1']),

