Index: unstable/setup.py
===================================================================
--- unstable.orig/setup.py
+++ unstable/setup.py
@@ -44,6 +44,7 @@ setup(
                 "loggerhead/controllers",
                 "loggerhead/middleware",
                 "loggerhead/templates",
+                "loggerhead/tests",
                 "breezy.plugins.loggerhead"],
     package_dir={'breezy.plugins.loggerhead':'.'},
     package_data = {"loggerhead": ["templates/*.pt",
