=== modified file '__init__.py'
--- a/__init__.py
+++ b/__init__.py
@@ -45,12 +45,6 @@
         (DEFAULT_PORT,))
 
 
-def load_loggerhead():
-    try:
-        from . import loggerhead
-    except ImportError:
-        import loggerhead
-
 def serve_http(transport, host=None, port=None, inet=None, client_timeout=None):
     # TODO: if we supported inet to pass requests in and respond to them,
     #       then it would be easier to test the full stack, but it probably
@@ -59,11 +53,10 @@
     from paste.httpexceptions import HTTPExceptionHandler
     from paste.httpserver import serve
 
-    load_loggerhead()
-    from .loggerhead.apps.http_head import HeadMiddleware
-    from .loggerhead.apps.transport import BranchesFromTransportRoot
-    from .loggerhead.config import LoggerheadConfig
-    from .loggerhead.main import setup_logging
+    from loggerhead.apps.http_head import HeadMiddleware
+    from loggerhead.apps.transport import BranchesFromTransportRoot
+    from loggerhead.config import LoggerheadConfig
+    from loggerhead.main import setup_logging
 
     if host is None:
         host = DEFAULT_HOST
@@ -95,8 +88,7 @@
     takes_args = ["filename"]
 
     def run(self, filename):
-        load_loggerhead()
-        from .loggerhead.loggerhead import load_test
+        from loggerhead.loggerhead import load_test
         script = load_test.run_script(filename)
         for thread_id in sorted(script._threads):
             worker = script._threads[thread_id][0]
@@ -107,7 +99,6 @@
 commands.register_command(cmd_load_test_loggerhead)
 
 def load_tests(loader, basic_tests, pattern):
-    load_loggerhead()
-    from .loggerhead.tests import test_suite
+    from loggerhead.tests import test_suite
     basic_tests.addTest(test_suite())
     return basic_tests
--- a/setup.py
+++ b/setup.py
@@ -38,6 +38,7 @@
                 "loggerhead/controllers",
                 "loggerhead/middleware",
                 "loggerhead/templates",
+                "loggerhead/tests",
                 "breezy.plugins.loggerhead"],
     package_dir={'breezy.plugins.loggerhead':'.'},
     package_data = {"loggerhead": ["templates/*.pt"]},
