=== modified file 'loggerhead/apps/__init__.py'
--- old/loggerhead/apps/__init__.py	2018-10-20 11:09:30 +0000
+++ new/loggerhead/apps/__init__.py	2018-11-27 20:18:10 +0000
@@ -6,8 +6,7 @@
 
 from ..util import convert_file_errors
 
-static = os.path.join(
-    os.path.dirname(os.path.dirname(__file__)), 'static')
+static = '/usr/share/loggerhead-breezy/static'
 
 # Static things can be cached for half a day, we could probably make this
 # longer, except for just before rollout times.

