=== modified file 'loggerhead/apps/__init__.py'
Index: breezy-unstable/loggerhead/apps/__init__.py
===================================================================
--- breezy-unstable.orig/loggerhead/apps/__init__.py
+++ breezy-unstable/loggerhead/apps/__init__.py
@@ -6,8 +6,7 @@ from paste import urlparser, fileapp
 
 from ..util import convert_file_errors
 
-static = os.path.join(
-    os.path.dirname(os.path.dirname(__file__)), 'static')
+static = '/usr/share/loggerhead-breezy/static'
 
 # Static things can be cached for half a day, we could probably make this
 # longer, except for just before rollout times.
