=== modified file 'loggerhead/__init__.py'
--- old/loggerhead/__init__.py	2022-08-21 12:50:04 +0000
+++ new/loggerhead/__init__.py	2022-08-21 13:25:06 +0000
@@ -26,6 +26,9 @@
     __version__ = importlib_metadata.version("loggerhead")
 except importlib_metadata.PackageNotFoundError:
     # Support running tests from the build tree without installation.
-    __version__ = None
+    import os
+    from setuptools.config import read_configuration
+    cfg = read_configuration(os.path.join(os.path.dirname(__file__), '..', 'setup.cfg'))
+    __version__ = cfg['metadata']['version']
 __revision__ = None
 required_breezy = (3, 1)

